test/promql/tests/application_errors_test.yaml

```
@@ -0,0 +1,108 @@
+evaluation_interval: 1m
+
+rule_files:
+  - prometheus.application_alerts.yaml
+
+tests:
+  # Alerted cases:
+  - interval: 5m
+    input_series:
+
+      # HAS is experiencince Application deletion failure rate of 100%, so it will be alerted
+      - series: 'has_application_successful_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '0'
+      - series: 'has_application_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1 2 3'
+      - series: 'has_application_failed_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1 2 3'
+
+    alert_rule_test:
+      - eval_time: 10m
+        alertname: ApplicationDeletionErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: application-service
+              pod: has
+              source_cluster: cluster01
+            exp_annotations:
+              summary: >-
+                HAS is experiencing application deletion failures of >1%
+              description: >-
+                Application controller in Pod has for namespace
+                application-service on cluster cluster01 is failing to
+                successfully delete at least 99% of applications over the past hour
+              runbook_url: TBD
+
+  - interval: 5m
+    input_series:
+
+      # HAS experienced 10% deletion failure rate, should still alert
+      - series: 'has_application_successful_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '0+1x90'
+      - series: 'has_application_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1+1x99'
+      - series: 'has_application_failed_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '0+1x10'
+
+    alert_rule_test:
+      - eval_time: 10m
```

I think this should be asserted after 1 hour of window has accumulated, and 5 minutes of evaluation timeout were reached. 
```suggestion
      - eval_time: 65m
```